new.go

```diff
@@ -51,6 +51,10 @@ func (*ZapPlugin) newWithChunkMode(results []index.Document,
 	chunkMode uint32) (segment.Segment, uint64, error) {
 	s := interimPool.Get().(*interim)
 
+	// this can be a reused interim object, so need to reset the
+	// bytes written stat count
+	s.setBytesWritten(0)
```

Comment:
This better placed inside the (s *interim) reset() method?

Reply:
Done